src/lib.rs

```diff
@@ -14,28 +16,33 @@ use crate::compile::compile_cdp_json;
 
 pub fn init() {
     const CDP_COMMIT: &str = "15f524c8f5ce5b317ddcdf5e6f875d6eb8bdac88";
+    let out_dir = env::var_os("OUT_DIR").unwrap();
     let mut file = OpenOptions::new()
         .append(true)
         .create(true)
-        .open(format!("./src/protocol.rs"))
+        .open(Path::new(&out_dir).join("protocol.rs"))
```

Comment:
wouldnt this code fail in normal compilations since OUT_DIR variable wont be set, correct me if im wrong

Reply:
It's set by cargo when building so it's always present. I missed the update on rustfmt below this, adding it in a second.

Reply:
Here's the documentation for the OUT_DIR https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts

Reply:
> It's set by cargo when building so it's always present. I missed the update on rustfmt below this, adding it in a second.

no i mean when the user builds this crate that env variable wont be set

Reply:
When you run `cargo build` the build.rs gets ran. Before executing, cargo sets the OUT_DIR environment variable as described in the link above.

Reply:
> When you run `cargo build` the build.rs gets ran. Before executing, cargo sets the OUT_DIR environment variable as described in the link above.

ah i understand now my brain didnt get it for some reason

Reply:
No worries, I'm on holiday mode as well but have been banging my head against the build script thingy so much recently that it still seems somewhat clear :D